pkg/sources/source_manager.go

```diff
@@ -368,37 +367,31 @@ func (api *headlessAPI) GetJobID(ctx context.Context, id int64) (int64, error) {
 
 // mgrUnitReporter implements the UnitReporter interface.
 type mgrUnitReporter struct {
-	unitCh       chan SourceUnit
-	unitErrs     []error
-	unitErrsLock sync.Mutex
+	unitCh chan SourceUnit
+	report *JobReport
```

Comment:
a thought: holding a reference to an entire report here (and also down in `mgrChunkReporter`) might be unnecessarily tight coupling, given all the other stuff in a `JobReport`. you could separate them with an interface on `JobReport` that just accepts errors - at the cost of adding another abstraction. what do you think?

Reply:
yeah, I think that will help define api expectations (which I'm still trying to figure out tbh).

Reply:
so this still looked weird to me, and i realized that it's because `JobReport` isn't really a job "report" anymore. it's, like, job runtime data? or something? `InProgressJob`?